pkg/reconciler/ingress/ingress.go

```diff
@@ -249,38 +247,37 @@ func (r *Reconciler) reconcileCertSecrets(ctx context.Context, ia *v1alpha1.Ingr
 }
 
 func (r *Reconciler) reconcileVirtualServices(ctx context.Context, ia *v1alpha1.Ingress,
-	desired []*v1alpha3.VirtualService) error {
+	desired map[string]*v1alpha3.VirtualService) error {
+	// Sort VirtualService names for a consistent processing order.
+	names := sets.NewString()
+	for name := range desired {
+		names.Insert(name)
+	}
 	// First, create all needed VirtualServices.
-	kept := sets.NewString()
-	for _, d := range desired {
-		if _, err := istioaccessor.ReconcileVirtualService(ctx, ia, d, r); err != nil {
+	for _, name := range names.List() {
+		vs := desired[name]
+		if vs == nil {
```

Comment:
The comment says `create all needed` but it seems you're going over the `nil` ones?

Reply:
Right, we change the map to have `non-nil` as to-be-created  and `nil` for to-be-deleted.  I updated the comments to clarify.

Reply:
Ah, i'd perceive the other way around :)